src/main/java/org/openmrs/maven/plugins/utility/DistroHelper.java

```
@@ -0,0 +1,254 @@
+
+    /**
+     * The Maven BuildPluginManager component.
+     */
+    BuildPluginManager pluginManager;
+
+    public DistroHelper(MavenProject mavenProject, MavenSession mavenSession, BuildPluginManager pluginManager) {
+        this.mavenProject = mavenProject;
+        this.mavenSession = mavenSession;
+        this.pluginManager = pluginManager;
+    }
+
+    /**
+     * @return distro properties from openmrs-distro.properties file in current directory or null if not exist
+     */
+    public static DistroProperties getDistroPropertiesFromDir() {
+        File distroFile = new File(new File(System.getProperty("user.dir")), "openmrs-distro.properties");
+        return getDistroPropertiesFromDir(distroFile);
+    }
+
+    /**
+     * @param distroFile file which contains distro properties
+     * @return distro properties loaded from specified file or null if file is not distro properties
+     */
+    public static DistroProperties getDistroPropertiesFromDir(File distroFile) {
+        if(distroFile.exists()){
+            try {
+                return new DistroProperties(distroFile);
+            } catch (MojoExecutionException e) {
+                return null;
+            }
+        }
+        else {
+            return null;
+        }
+    }
+
+    /**
+     * valid formats are 'groupId:artifactId:version' and 'artifactId:version'
+     * parser makes user-friendly assumptions, like inferring default groupId or full artifactId for referenceapplication
+     * returns null if string is invalid
+     */
+    public static Artifact parseDistroArtifact(String distro){
+        String[] split = distro.split(":");
+        if(split.length == 2){
+            return new Artifact(inferDistroArtifactId(split[0], Artifact.GROUP_DISTRO), split[1], Artifact.GROUP_DISTRO);
+        } else if (split.length == 3){
+            return new Artifact(inferDistroArtifactId(split[1], split[0]), split[2], split[0]);
+        } else {
+            return null;
```

Please throw an exception here instead.